src/Raven.Server/Documents/Queries/Suggestions/SuggestionsQueryServerSide.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Http;
+using Raven.Client.Documents.Queries;
+using Raven.Client.Documents.Queries.Suggestion;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.Documents.Queries.Suggestions
+{
+    public class SuggestionsQueryServerSide : IIndexQuery
+    {
+        public static float DefaultAccuracy = 0.5f;
+
+        public static int DefaultMaxSuggestions = 15;
+
+        public static StringDistanceTypes DefaultDistance = StringDistanceTypes.Levenshtein;
+
+        /// <summary>
+        /// Create a new instance of <seealso cref="SuggestionQuery"/>
+        /// </summary>
+        public SuggestionsQueryServerSide()
+        {
+            Accuracy = 0.5f;
+            Distance = DefaultDistance;
+            MaxSuggestions = DefaultMaxSuggestions;
+        }
+
+
+        public static SuggestionsQueryServerSide Create(HttpContext httpContext, int pageSize, JsonOperationContext context)
+        {
+            var result = new SuggestionsQueryServerSide
+            {
+                // all defaults which need to have custom value
+                PageSize = pageSize,                
+            };
+
+            foreach (var item in httpContext.Request.Query)
```

Comment:
IIRC this was done like this because performance was better when loop was used then if we searched for each element individually.